### PR TITLE
Geas plugin

### PR DESCRIPTION
**Pull request by crownedgrouse**
Geas plugin for erlang.mk using new external plugin way.

### CODE REVIEW

**Comment by essen** [comment]
Cool! Will try when I can, thanks for your patience in advance though. :-)

**Comment by crownedgrouse** [comment]
Not a problem with external plugin. I can use it anyway already. Looks like CI tests fails, but seems general to all almost other PR.

File: test/plugin_geas.mk [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+GEAS_CASES = app 
+GEAS_TARGETS = $(addprefix geas-,$(GEAS_CASES))
+GEAS_CLEAN_TARGETS = $(addprefix clean-,$(GEAS_TARGETS))
+
+.PHONY: geas clean-geas $(GEAS_CLEAN_TARGETS)
+
+clean-geas: $(GEAS_CLEAN_TARGETS)
+
+$(GEAS_CLEAN_TARGETS):
+	rm -rf $(APP_TO_CLEAN)/
```
Comment (crownedgrouse) [inline test/plugin_geas.mk]: Loïc, I did like other tests , but I think this line if dangerous in all sibling makefiles. Let imagine that $(APP_TO_CLEAN) is empty ...  I suppose something like  : test -z "$(APP_TO_CLEAN)" || rm -rf $(APP_TO_CLEAN)/
is safer ... MHO

Comment (essen) [inline test/plugin_geas.mk]: Better just drop the / at the end. Then it'll error out if there's no such file.

Comment (crownedgrouse) [inline test/plugin_geas.mk]: better idea , yes.  'rm -rf ' alone do nothing and return 0, so no error.

Comment (essen) [inline test/plugin_geas.mk]: Can you open a new ticket for this? Or a PR if you like. I'd like to solve it as soon as possible, to avoid any issues.

Comment (crownedgrouse) [inline test/plugin_geas.mk]: As far plugins tests may change in close futur, doing a PR for all sibling Makefile is maybe not a good idea. I will amend mine for the moment.

Comment (crownedgrouse) [inline test/plugin_geas.mk]: sorry posts are crossing. I will amend mine and open a ticket for others. If I have some time, will fix and propose a PR for others files.

**Comment by crownedgrouse** [comment]
just updated my branch with the rm -rf /  fix.  I squashed the fix in last commit.

**Comment by crownedgrouse** [comment]
I did not notice it before, but my pluging.mk 'help::'  target is not add to end of 'make help' output.
A bug I suppose.

**Comment by essen** [comment]
Perhaps it's in the middle or the beginning? If it's a bug open a separate ticket. :-)

**Comment by crownedgrouse** [comment]
I double check and you are right. There is however a difference between 'make help' in geas repository and same command in another repository where geas plugin is used. First give help in bottom of help, while second give help just after 'Core targets' help. The reason of my wrong guess.
I do not know if you consider it as a bug or not.

**Comment by crownedgrouse** [comment]
sorry, my bad. In geas repository, I included plugins.mk after erlang.mk. Obviously I got help at bottom.

**Comment by essen** [comment]
Should create a target "plugins-help" or something that is called at the end. Open a new ticket. :-)

**Comment by essen** [comment]
Hey, for now I will not merge this, and limit myself to listing external plugins, with a link to the project. This leaves responsibility of documenting/testing the plugin to the plugin author. In the future I may revisit this decision and/or improve the interface for discovering docs and/or tests, but right now I have to make decisions and move on. I think the docs/tests in this PR still would be great in the geas project (if they're not there yet already).

Closing, thanks! I will push the new chapter with the plugins list in a few minutes. Patches to it are of course welcome if I missed something.

**Comment by crownedgrouse** [comment]
I was almost sure this would end to this decision. External plugins are owners responsability. However a need to include plugins help nicely in the 'make help' remains. Will add the documentation in my own wiki.

**Comment by essen** [comment]
Definitely, the other ticket will be worked on to improve this. Thanks for understanding.

**Comment by crownedgrouse** [comment]
no worries, it was just a reminder.

**Comment by crownedgrouse** [comment]
and feel free to ask for help to some contributors. Some subjects may be handled by some of us. We simply need to know if you want to handle some needs by your own. You certainly have to work for money and opensource need lot of time. I read you post on this ... :>)

**Comment by essen** [comment]
I'm still going toward the goal of documenting/testing anything. Other issues are free to grab. :-)

**Comment by crownedgrouse** [comment]
you might add two labels to tag some issues, one to reserve the issue to yourself for any reasons, or on contrary call for contributors to pick-up it. Once picked-up , set the 'reserved' label on it. Nobody want to work for nothing or in dooblon with another. This may emulate people that like your nice project...

**Comment by essen** [comment]
That would be what "assigned" is for, and right now I have no issue assigned.

**Comment by crownedgrouse** [comment]
https://github.com/ninenines/erlang.mk/labels do not show such label, so that puzzled me. I suppose only label used are shown. Never noticed that.

**Comment by essen** [comment]
Look at the top right of tickets or even https://github.com/ninenines/erlang.mk/issues - Assignee.

I don't use that, but if I did make known what I currently work on that's what I would use. I don't use it because I tend to work on different things at the same time, some very slowly. And if there is duplication I tend to accept a PR and add my changes on top of it during the merge.

**Comment by crownedgrouse** [comment]
OK, so forget my former comments. This may be useful to use it, but need someone to assign ;-)

**Pull request closed without merging**